### ISSUE TITLE
Warnings generated by units tests 

### ISSUE THREAD

**Issue by GernotMaier** [gammasim/simtools#510]
Running unit tests produce the following warnings. In an ideal world, we wouldn't see any warnings:

```
========================================================================================== warnings summary ==========================================================================================
tests/unit_tests/test_simulator.py::test_make_resources_report
  /usr/local/lib/python3.10/dist-packages/numpy/core/fromnumeric.py:3504: RuntimeWarning: Mean of empty slice.
    return _methods._mean(a, axis=axis, dtype=dtype,

tests/unit_tests/test_simulator.py::test_make_resources_report
  /usr/local/lib/python3.10/dist-packages/numpy/core/_methods.py:129: RuntimeWarning: invalid value encountered in scalar divide
    ret = ret.dtype.type(ret / rcount)

tests/unit_tests/corsika/test_corsika_output_visualize.py::test_kernel_plot_1D_photons
  /workdir/external/simtools/simtools/corsika/corsika_output_visualize.py:238: RuntimeWarning: More than 20 figures have been opened. Figures created through the pyplot interface (`matplotlib.pyplot.figure`) are retained until explicitly closed and may consume too much memory. (To control this warning, see the rcParam `figure.max_open_warning`). Consider using `matplotlib.pyplot.close()`.
    fig, ax = plt.subplots()

tests/unit_tests/utils/test_general.py::test_convert_2D_to_radial_distr
  /workdir/external/simtools/simtools/util/general.py:852: RuntimeWarning: invalid value encountered in scalar divide
    histogram_1D[i_radial] = np.sum(hist_sorted[indices_to_sum]) / weights[i_radial]

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

Some of them are related to the simulated (@orelgueta), some to the corsika output (@VictorBarbosaMartins )

**Comment by orelgueta** [comment]
We always had a warning on something deprecated so I didn't notice these new ones. I will fix them (and the deprecated ones).

**Comment by VictorBarbosaMartins** [comment]
I agree there should be no warnings being raised in tests. @orelgueta, please let me know when you are done and I will fix the corsika_output part

**Comment by orelgueta** [comment]
This is solved in #591. There's a new warning now, but it's the result of a specific version of matplotlib, so I suggest to close this and hope that it is resolved by them soon. If not, we can open a new issue in the future to deal with.
